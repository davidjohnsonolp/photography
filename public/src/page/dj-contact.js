import buildShadowRoot from '../lib/build-shadow-root.js';

class DJContact extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
          display: flex;
          flex-direction: column;
          flex: 1 0 auto;
        }
        main {
          background: white;
          overflow: hidden;
          flex: 1 0 auto;
          padding: 0 5px;
        }
      </style>
      <dj-header>
      </dj-header>

      <main>
        <dj-h2>Contact</dj-h2>
      </main>

      <dj-footer>
      </dj-footer>
    `;
    buildShadowRoot(html,this);
  }
}

customElements.define('dj-contact', DJContact);
export default DJContact;
