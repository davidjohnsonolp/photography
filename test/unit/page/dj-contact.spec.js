import '../../../public/src/page/dj-contact.js';

let contactPage;

describe('DJContact Contact Page', () => {
  beforeEach(() => {
    contactPage = window.document.createElement('dj-contact'); 
    document.body.appendChild(contactPage);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-contact')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: contactPage.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});
