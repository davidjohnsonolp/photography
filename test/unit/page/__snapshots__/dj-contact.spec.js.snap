// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJContact Contact Page renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
          display: flex;
          flex-direction: column;
          flex: 1 0 auto;
        }
        main {
          background: white;
          overflow: hidden;
          flex: 1 0 auto;
          padding: 0 5px;
        }
      </style>
      <dj-header>
      </dj-header>

      <main>
        <dj-h2>Contact</dj-h2>
      </main>

      <dj-footer>
      </dj-footer>
    ",
}
`;
